### PR TITLE
Added SASS to the UI and build workflow

### PR DESCRIPTION
**Pull request by hortha**
### Context

We need a way to define styles that can use variables from the base GDS library as well as, potentially, importing components from other design systems e.g. MOJ.

### Changes proposed in this pull request

Add SASS to the SupportUi in the same way it’s set up in [GitHub - DFE-Digital/get-an-identity](https://github.com/DFE-Digital/get-an-identity)

The project file, build workflow, docs will all need updating. Use libman to reference the GDS library.

### Guidance to review

N/A

### Checklist

-   [x] Attach to Trello card
-   [ ] Rebased master
-   [ ] Cleaned commit history
-   [x] Tested by running locally

### CODE REVIEW

**Comment by hortha** [comment]
[Add SASS to TRS Console](https://trello.com/c/iVBjbxh3/1199-add-sass-to-trs-console)

File: justfile [hunk 1]
```diff
@@ -80,6 +81,7 @@ docker-build-cli *ARGS:
 
 # Build the Support UI Docker image
 docker-build-ui *ARGS:
+  @cd {{solution-root}} && npm install -g sass
```
Comment (gunndabad) [inline justfile:84]: I don't think we need this here.

File: justfile [hunk 2]
```diff
@@ -8,6 +8,7 @@ default:
 # Install .NET local tools
```
Comment (gunndabad) [inline justfile:8]:
```suggestion
# Install local tools
```

File: justfile [hunk 3]
```diff
@@ -8,6 +8,7 @@ default:
...
 install-tools:
   @cd {{solution-root}} && dotnet tool restore
+  @cd {{solution-root}} && npm install -g sass
```
Comment (gunndabad) [inline justfile:11]:
```suggestion
  npm install -g sass 
```

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/wwwroot/Styles/site.scss [hunk 4]
```diff
@@ -1,3 +1,5 @@
+
+
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/wwwroot/Styles/site.scss:2]: Can we remove the `.css` file from git too?

File: justfile (hunk 1 above)
Comment (hortha) [reply justfile:84]: we do as the initial PR failed

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/TeachingRecordSystem.SupportUi.csproj [hunk 5]
```diff
@@ -5,6 +5,10 @@
     <UserSecretsId>TeachingRecordSystemSupportUi</UserSecretsId>
   </PropertyGroup>
 
+  <Target Name="CompileSass" BeforeTargets="Build">
+    <Exec Command="sass -q wwwroot/Styles/site.scss wwwroot/Styles/site.css" />
+  </Target>
+
   <ItemGroup>
     <PackageReference Include="AspNetCore.HealthChecks.NpgSql" Version="7.0.0" />
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/TeachingRecordSystem.SupportUi.csproj:13]: Think we need a `Microsoft.Web.LibraryManager.Build` package reference too

File: justfile (hunk 1 above)
Comment (gunndabad) [reply justfile:84]: ID installs SASS explicitly before doing the build: https://github.com/DFE-Digital/get-an-identity/blob/main/.github/workflows/build.yml#L89

Comment (hortha) [reply justfile]: Yeah but it does the build publish etc in the same workflow as the docker build - it is done in totally separate workflow in TRS in the package.yml

Comment (hortha) [reply justfile]: I did add steps to install SASS in other parts of the build so I was also suprised it failed until I realised why!

File: justfile (hunk 2 above)
Comment (hortha) [reply justfile]: Done

File: justfile (hunk 3 above)
Comment (hortha) [reply justfile]: Done

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/wwwroot/Styles/site.scss (hunk 4 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/wwwroot/Styles/site.scss:2]: Already done

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/TeachingRecordSystem.SupportUi.csproj (hunk 5 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/TeachingRecordSystem.SupportUi.csproj:13]: Yep - didn't realise until reading libman docs - done now

**Review by gunndabad — APPROVED** [review]

**Pull request merged**